src/modules/main/store/mainReducer.ts

```diff
@@ -1,18 +1,22 @@
 import { IPost, IExpert } from '../../../lib/types';
-import {
-  MainActions,
-  INewestItem,
-  LoadData,
-} from './actionTypes';
+import { MainActions, LoadData, INewestPosts } from './actionTypes';
 
 export interface IMainState {
-  newest: INewestItem[];
+  newest: INewestPosts;
   important: IPost[];
   experts: IExpert[];
 }
 
 const initialState: IMainState = {
-  newest: [],
+  newest: {
+    posts: [],
+    meta: {
+      limit: 0,
+      totalNewestPosts: 50,
```

Comment:
should it be here?

Reply:
Unfortunately yes, without totalNewestPosts: 50 initialization, button showMore not working properly.

Reply:
so again,
what are these for:
`limit` - ?
`totalNewestPosts` - ?

Reply:
fixed
https://github.com/ita-social-projects/dokazovi-fe/pull/39/commits/ac3df2f0c2f7903c7ded51f35783de099ec59c09
For now, we don need to initiate state with 
```
limit: 0,
totalNewestPosts: 50,
showMore: true,
```

I have added a check for initial dispatch `isTotalNewPosts`. 
Also, with this check, we can work with mock-data of any length with no need to hardcode it in the config file.